index.php

```diff
@@ -0,0 +1,319 @@
+<?php
+
+namespace VUMC\REDCapInstrumentAndVariableSQLRenamer;
+
+$pid = (int)$_GET['pid'];
+if (!array_key_exists("U", $_REQUEST) && $_REQUEST['message'] != "U") {
+    $_SESSION['message'] = "";
+    $_SESSION['message_type'] = "";
+}
+?>
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <meta name="description" content="">
+    <meta name="author" content="">
+    <meta http-equiv="Cache-control" content="public">
+    <meta name="theme-color" content="#fff">
+    <link type='text/css' href='<?= $module->getUrl('css/styles.css') ?>' rel='stylesheet' media='screen'/>
+    <?php
+    echo $module->loadREDCapJS(); ?>
+    <script>
+        $(document).ready(function () {
+            $(function () {
+                $('[data-toggle="tooltip"]').tooltip();
+            });
+
+            $('.autocomplete-input').keyup(function () {
+                let term = $(this).val();
+                if (term === "") {
+                    $("#new_name_input").hide();
+                }
+                getAutocompleteData(term, "old_var");
+            });
+
+            $('[name=data_type]').change(function () {
+                if ($("#input-data-old").val() !== "") {
+                    getAutocompleteData($("#input-data-old").val(), "old_var");
+                } else {
+                    $("#new_name_input").hide();
+                }
+            });
+
+            $(document).mouseup(function (e) {
+                var container = $(".autocomplete-items");
+
+                // if the target of the click isn't the container nor a descendant of the container
+                if (!container.is(e.target) && container.has(e.target).length === 0) {
+                    container.hide();
+                }
+                if ($("#input-data-old").val() === "") {
+                    $("#new_name_input").hide();
+                }
+            });
+
+            $('#btnConfirm').on("click", function (e) {
+                let old_var = $('#input-data-old').val();
+                let new_var = $('#input-data-new').val();
+                saveData("<?=$pid?>", "instrument", old_var, new_var);
+            });
+
+            $('input, button').on("click", function (e) {
+                $(".autocomplete-start").hide();
+                // $(e.target).hasClass('autocomplete-input')
+                if ($(this).attr('autocomplete') == "button" || ($(this).attr('autocomplete') == "input" && $('.autocomplete-input').val() === "")) {
+                    let type = $('input[name="data_type"]:checked').attr('id');
+                    $("#select-" + type).show();
+                }
+            });
```

Comment:
Consider using an existing autocomplete front end library for this.  Either jQuery UI or select2.  My vote is select2 because you can configure the data handler to use your ajax: https://select2.org/data-sources/ajax#transforming-response-data

Reply:
Jquery UI also allows for a remote data handler: [https://api.jqueryui.com/autocomplete/#option-source](https://api.jqueryui.com/autocomplete/#option-source).